notebook/services/api/api.yaml

```diff
@@ -654,12 +639,18 @@ definitions:
       id:
         type: string
         format: uuid
-      notebook:
-        type: object
-        properties:
-          path:
-            type: string
-            description: path to notebook
+      path:
+        type: string
+        description: "The path to the session.  For console sessions, the path 
+         contains the cwd of the kernel and the name of the console session, 
+         which may be a uuid"
+      type:
+        type: string;
+        description: session type
+        enum:
+          - notebook
+          - file
+          - console
```

Comment:
The filesystem knowledge is already in the API, since we are starting a kernel in a specified directory.  The fact that it is relative shouldn't matter.